### PR TITLE
added pagination for arguments

### PR DESCRIPTION
**Pull request by purple-void**
we also added some comments

### CODE REVIEW

**Review by Monstarrrr — CHANGES_REQUESTED** [review]
1. Why are we making pagination for arguments specifically? Isn't pagination the same logic regardless of the type of content?  
Use django setting to apply to any endpoint  

2. The front-end should be able to decide on how many items per page there are. Do not hard-code it in the back-end unless as a default value

File: backend/.template-env [hunk 1]
```diff
@@ -29,6 +29,8 @@ POSTGRES_USER=
 POSTGRES_PASSWORD=
 POSTGRES_HOST=
 
+PAGE_SIZE=
```
Comment (seporterfield) [inline backend/.template-env:32]: missing short comment

File: backend/rebutify/settings.py [hunk 2]
```diff
@@ -94,6 +94,10 @@
         "rest_framework_simplejwt.authentication.JWTAuthentication",
     ],
     "DEFAULT_SCHEMA_CLASS": "drf_spectacular.openapi.AutoSchema",
+    # https://stackoverflow.com/a/47657610/19071246
+    # cursor pagination uses previous or next page links
+    "DEFAULT_PAGINATION_CLASS": "rest_framework.pagination.CursorPagination",
+    "PAGE_SIZE": os.getenv("PAGE_SIZE", 10),
```
Comment (seporterfield) [inline backend/rebutify/settings.py:100]: wrap with `int()` type conversion since os.getenv will return a string if the value is set

**Review by seporterfield — CHANGES_REQUESTED** [review]
commented 2 changes
looks great and nice job!

**Comment by Monstarrrr** [comment]
> @Monstarrrr we changed pagination to apply globally. PAGE_SIZE is set in env. Does that work for the front-end?

The page size should be part of the request body or parameters e.g. ``api/posts?size=15&page=1``

**Review by seporterfield — APPROVED** [review]

**Comment by seporterfield** [comment]
> @seporterfield we think the checks are wrong. we can't reproduce the error in our code

did you read the error/output of the checks?

**Comment by seporterfield** [comment]
The way we're parsing .env variables has a side effect of any empty variables in .env being parsed as the empty string instead of `None` which is kinda annoying but I haven't been able to find a solution that doesn't make things even harder for people trying to use our project.

Fix option 1:
Put `sed -i 's/^PAGE_SIZE=$/PAGE_SIZE=20/' .env` in the "Write dummy .env" step in the workflow file. This is fitting for variables that need to be changed to something for production that is different than the default that might be provided in .template-env

Fix option 2:
Make `PAGE_SIZE=20` in .template-env
This is fitting if we want a page size of 20 by default for anyone setting up their own instance of Rebutify.

**Comment by seporterfield** [comment]
> We fixed it for now with `"PAGE_SIZE": int(os.getenv("PAGE_SIZE", "10") or "10")`

this is probably the best long term strategy for non-string env vars, looks great

**Review by seporterfield — APPROVED** [review]

**Review by Monstarrrr — APPROVED** [review]
To the rescue

**Pull request merged**